composer/loggers/mosaicml_logger.py

```diff
@@ -88,14 +88,58 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> No
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # Log model data downloaded and initialized for run events
+        self._log_metadata({'model_initialized_time': time.time()})
         # Log WandB run URL if it exists. Must run on after_load as WandB is setup on event init
         for callback in state.callbacks:
             if isinstance(callback, WandBLogger):
                 run_url = callback.run_url
                 if run_url is not None:
                     self._log_metadata({'wandb/run_url': run_url})
 
+    def _get_training_progress_metrics(self, state: State) -> Dict[str, Any]:
```

Comment:
nit: Move this function outside the class because it doesn't rely on `self` at all 😃

Reply:
this does need `self` since dataloader len should be cached (see logic from runtime estimator). missed in review but this comment reminded me :)